src/BuildScriptGenerator/BuildScriptGeneratorContext.cs

```diff
@@ -15,34 +15,6 @@ public partial class BuildScriptGeneratorContext : RepositoryContext
         /// </summary>
         public string OperationId { get; set; }
 
-        /// <summary>
```

Comment:
Following is the basic idea of this PR:

- We want to enable users to provide settings through
   - command line
   - environment variables
   - build.env file
   - app.yml in future

- There are some settings which (for now) I am not including as part of these settings. They are like 'source dir', 'destination dir' etc. This is because it does not make sense for an end user to provide these.

- I deleted properties here because we want to access these settings via `options` directly. Remember that values in options are 'realized' ones, i.e when you say `options.EnableDynamicInstall`, you are guaranteed that the value returned is hierarchically considered one.